src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -330,6 +320,25 @@ private Map<Integer, Float> doANNSearch(final LeafReaderContext context, final B
             .collect(Collectors.toMap(KNNQueryResult::getId, result -> knnEngine.score(result.getScore(), spaceType)));
     }
 
+    @VisibleForTesting
+    List<String> getEngineFiles(SegmentReader reader, String extension) throws IOException {
```

Comment:
1. Shouldn't we make this method to return `Path indexPath` of the engine file?
2. Is there a way to move this method outside of KNNWeight so that we can avoid `@VisibleForTesting`?

Reply:
> Shouldn't we make this method to return Path indexPath of the engine file?

i prefer to return List<String>,  there is some other engines we may imported in the future, we can not assume that we only have one path.

Reply:
> Is there a way to move this method outside of KNNWeight so that we can avoid @VisibleForTesting?

how about move method into KNNCodecUtil as a static method?

Reply:
The main motive of moving the method out of this class is to make it easier to write a unit test of the caller of this method. In that sense, no much difference between current code and static method. I am fine with current code as it might need non trivial change to do that...

Reply:
> > Shouldn't we make this method to return Path indexPath of the engine file?
> 
> i prefer to return List, there is some other engines we may imported in the future, we can not assume that we only have one path.

We are already assuming that there is only one path if you the next code where use only use the first value from the list. 
```
Path indexPath = PathUtils.get(directory, engineFiles.get(0));
```

Reply:
> We are already assuming that there is only one path. If you see the following code, it uses only the first value from the list.

it is because we using one file right now, but i say in the future we can have multi files, and we can only changed this, without change getEngineFiles